src/enumerator/arithmetic_sequence_object.cpp

```diff
@@ -0,0 +1,261 @@
+#include "natalie/enumerator/arithmetic_sequence_object.hpp"
+#include "natalie.hpp"
+
+namespace Natalie::Enumerator {
+ArithmeticSequenceObject::ArithmeticSequenceObject(Env *env, Origin origin, Value begin, Value end, Value step, bool exclude_end)
+    : ArithmeticSequenceObject {} {
+    m_origin = origin;
+    m_begin = begin;
+    m_end = end;
+    m_exclude_end = exclude_end;
+    m_step = step;
+
+    auto Enumerator = GlobalEnv::the()->Object()->const_fetch("Enumerator"_s)->as_module();
+    auto method_info = Enumerator->find_method(env, "initialize"_s);
+    method_info.method()->call(env, this, {}, new Block { env, this, enum_block, 1 });
+}
+
+bool ArithmeticSequenceObject::calculate_ascending(Env *env) {
+    return step()->send(env, ">"_s, { Value::integer(0) })->is_truthy();
+}
+
+Integer ArithmeticSequenceObject::calculate_step_count(Env *env) {
+    if (!m_end || m_end->is_nil() || m_end->send(env, "infinite?"_s)->is_truthy())
+        return 0;
+
+    auto _step = step();
+    auto _begin = m_begin;
+    auto _end = m_end;
+
+    auto cmp = ascending(env) ? ">"_s : "<"_s;
+    if (m_begin->send(env, cmp, { m_end })->is_truthy())
+        return 0;
+
+    if (_step->is_float() || _begin->is_float() || _end->is_float()) {
+        _step = _step->to_f(env);
+        _begin = _begin->to_f(env);
+        _end = _end->to_f(env);
+    }
+
+    auto n = _end->send(env, "-"_s, { _begin })->send(env, "/"_s, { _step->to_f(env) });
```

Comment:
Oh yeah, I noticed you're using `->send()` a lot here. Let's try to use `.send()` instead, just for consistency. I'm afraid if we get in the habit of bypassing the `Value::send()` machinery, we'll lose some of our fast int/float optimizations accidentally.

(We should probably rename Object::send so it's not so easy to use directly.)